numba/targets/linalg.py

```diff
@@ -1110,3 +1110,285 @@ def qr_impl(a):
         return (q[:, :minmn], r)
 
     return qr_impl
+
+
+# helpers and jitted specialisations required for np.linalg.lstsq
+def _check_linalg_1_or_2d_matrix(a, func_name):
+    # checks that a matrix is 1 or 2D
+    if not isinstance(a, types.Array):
+        raise TypingError("np.linalg.%s() only supported for array types "
+                          % func_name)
+    if not a.ndim <= 2:
+        raise TypingError("np.linalg.%s() only supported on 1 and 2-D arrays "
+                          % func_name)
+    if not isinstance(a.dtype, (types.Float, types.Complex)):
+        raise TypingError("np.linalg.%s() only supported on "
+                          "float and complex arrays." % func_name)
+
+
+def _get_res_impl(dtype, real_dtype, b):
+    # gets an implementation for computing the residual
+    ndim = b.ndim
+    if ndim == 1:
+        if isinstance(dtype, (types.Complex)):
+            @jit(nopython=True)
+            def cmplx_impl(b, n, nrhs):
+                res = np.empty((1,), dtype=real_dtype)
+                res[0] = np.sum(np.abs(b[n:, 0])**2)
```

Comment:
Thanks. I am probably going to leave this unless it becomes performance damaging because it has similar logic to numpy. The method to get to this point is ~O(N^3) so I'd hope it doesn't have too much impact.